SFINCS v2.3.0 outputs SGRID for quadtree models — `hydromt_sfincs` cannot read it

### HydroMT-SFINCS version checks

- [x] I have checked that this issue has not already been reported.
- [x] I have checked that this bug exists on the latest version of HydroMT-SFINCS.

### Reproducible Example

## Setup  
1. **SFINCS**: v2.3.0 mt. Faber Release (2025-11-18) downloaded from the deltares website
2. **hydromt_sfincs @ main**: v2.0.0.dev0
3. built quadtree model with the fixture [`build_sfincs_model_quadtree`](https://github.com/Deltares-research/FloodAdapt-core/blob/952029d9ddacf70fd6b9945997822f1df10b2394/tests/conftest.py#L577)
4. Added some forcings to it (specifics not relevant, just testing the floodmap creation).
5. run SFINCS

## Steps to reproduce

1. Extract attached quadtree SFINCS model
2. Try to read the output:

```python
from hydromt_sfincs import SfincsModel
from pathlib import Path

path = Path(...)
mod = SfincsModel(path)
mod.output.read()
```

after 4: [before_running_sfincs.zip](https://github.com/user-attachments/files/26569789/before_running_sfincs.zip)
after 5: [after_running_sfincs.zip](https://github.com/user-attachments/files/26569725/after_running_sfincs.zip)

Error:
```text
flood_adapt\adapter\sfincs_adapter.py:1747: in _get_zsmax
    self._model.output.read()
..\hydromt_sfincs\hydromt_sfincs\components\output.py:85: in read
    self.read_map_file(
..\hydromt_sfincs\hydromt_sfincs\components\output.py:149: in read_map_file
    with xu.load_dataset(fn_map) as ds:
         ^^^^^^^^^^^^^^^^^^^^^^^
.pixi\envs\default\Lib\site-packages\xugrid\core\common.py:30: in load_dataset
    return _dataset_helper(ds)
           ^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

ds = <xarray.Dataset> Size: 98kB
Dimensions:        (corner_n: 37, corner_m: 85, n: 36, m: 84, time: 1,
                   ...uild-Date:                          $Date: 2025-11-18
    title:                               SFINCS map netcdf output

    def _dataset_helper(ds: xr.Dataset):
        n_topology = len(ds.ugrid_roles.topology)
        if n_topology == 0:
>           raise ValueError(
                "The file or object does not contain UGRID conventions data. "
                "One or more UGRID topologies are required. Perhaps you wrote "
                "the file using `data.to_netcdf()` instead of `data.ugrid.to_netcdf()`?"
            )
E           ValueError: The file or object does not contain UGRID conventions data. One or more UGRID topologies are required. Perhaps you wrote the file using `data.to_netcdf()` instead of `data.ugrid.to_netcdf()`?

.pixi\envs\default\Lib\site-packages\xugrid\core\common.py:15: ValueError
```

### Current behaviour

The output file is SGRID with `(n, m)` dimensions:

```
Conventions: CF-1.6, SGRID-0.3
Dims:    n=36, m=84, time=1, timemax=1
Variables: zsmax(timemax, n, m), zs(time, n, m), zb(n, m), msk(n, m), ...
```

### Desired behaviour

UGRID format with `mesh2d_nFaces` dimension, consistent with the quadtree input file `sfincs.nc`:

```
Dims:    mesh2d_nFaces=3024
Variables: zsmax(timemax, mesh2d_nFaces), ...
```


### Additional context

`hydromt_sfincs`'s `SfincsOutput.read_map_file` assumes quadtree output is UGRID and calls `xu.load_dataset()`, which raises a `ValueError`. This blocks reading model results and any downstream processing (floodmap generation, etc.).

--- a/hydromt_sfincs/components/config/config_variables.py
+++ b/hydromt_sfincs/components/config/config_variables.py
@@ -841,10 +841,10 @@ class SfincsConfigVariables(BaseSettings):
         0, ge=0, le=1, description="Write dynamic bed level output (1: yes, 0: no)"
     )
     regular_output_on_mesh: int = Field(
-        0,
+        1,
         ge=0,
         le=1,
-        description="Write regular grid on quadtree mesh output (1: yes, 0: no)",
+        description="Write quadtree without refinement on quadtree mesh (1: yes) or regular grid (0: no)",
     )
     twet_threshold: float = Field(
         0.01, ge=0.0, description="Time-wet minimum depth threshold (m)"

